### ISSUE TITLE
Mysql 5.7 - Data too long for column 'type_os'

### ISSUE THREAD

**Issue by jajm** [omeka/Omeka#738]
Hello,

On Omeka 2.3.1 (with MySQL 5.7) I tried to upload a .jpg image and it failed with the following error:

```
2016-09-08T11:27:21+02:00 ERR (3): exception 'Zend_Db_Statement_Mysqli_Exception' with message 'Mysqli statement execute error : Data too long for column 'type_os' at row 1' in /home/omeka/www/application/libraries/Zend/Db/Statement/Mysqli.php:214
Stack trace:
#0 /home/omeka/www/application/libraries/Zend/Db/Statement.php(303): Zend_Db_Statement_Mysqli->_execute(Array)
#1 /home/omeka/www/application/libraries/Zend/Db/Adapter/Abstract.php(480): Zend_Db_Statement->execute(Array)
#2 [internal function]: Zend_Db_Adapter_Abstract->query('INSERT INTO `om...', Array)
#3 /home/omeka/www/application/libraries/Omeka/Db.php(79): call_user_func_array(Array, Array)
#4 /home/omeka/www/application/libraries/Omeka/Db.php(252): Omeka_Db->__call('query', Array)
#5 /home/omeka/www/application/libraries/Omeka/Db.php(252): Omeka_Db->query('INSERT
INTO `om...', Array)
#6 /home/omeka/www/application/libraries/Omeka/Record/AbstractRecord.php(543): Omeka_Db->insert('File', Array)
#7 /home/omeka/www/application/models/Item.php(335): Omeka_Record_AbstractRecord->save()
#8 /home/omeka/www/application/models/Item.php(248): Item->saveFiles()
#9 [internal function]: Item->afterSave(Array)
#10 /home/omeka/www/application/libraries/Omeka/Record/AbstractRecord.php(277): call_user_func(Array, Array)
#11 /home/omeka/www/application/libraries/Omeka/Record/AbstractRecord.php(550): Omeka_Record_AbstractRecord->runCallbacks('afterSave', Array)
#12 /home/omeka/www/application/libraries/Omeka/Controller/AbstractActionController.php(188): Omeka_Record_AbstractRecord->save(false)
#13 /home/omeka/www/application/controllers/ItemsController.php(151): Omeka_Controller_AbstractActionController->addAction()
#14 /home/omeka/www/application/libraries/Zend/Controller/Action.php(516): ItemsController->addAction()
#15 /home/omeka/www/application/libraries/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch('addAction')
#16 /home/omeka/www/application/libraries/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http))
#17 /home/omeka/www/application/libraries/Zend/Application/Bootstrap/Bootstrap.php(105): Zend_Controller_Front->dispatch()
#18 /home/omeka/www/application/libraries/Zend/Application.php(382): Zend_Application_Bootstrap_Bootstrap->run()
#19 /home/omeka/www/application/libraries/Omeka/Application.php(79): Zend_Application->run()
#20 /home/omeka/www/admin/index.php(28): Omeka_Application->run()
#21 {main}
```

files.type_os is VARCHAR(255) and apparently it's not sufficient. Maybe change the column type to TEXT ?

**Comment by zerocrates** [comment]
We fixed this in 2.4 in basically the reverse way: truncating over-long type_os values.

**Comment by jajm** [comment]
Ok, I confirm that it's fixed in 2.4.1, thanks
